pygfx/utils/transform.py

```diff
@@ -852,10 +862,13 @@ def matrix(self) -> np.ndarray:
     @matrix.setter
     def matrix(self, value: np.ndarray):
         if self._parent:
-            self.own.matrix = self._parent.inverse_matrix @ value
+            if self.own.state_basis == "matrix":
+                np.matmul(self._parent.inverse_matrix, value, out=self.own._matrix)
+                self.own.flag_update()
+            else:
+                self.own.matrix = self._parent.inverse_matrix @ value
         else:
             self.own.matrix = value
-        self.flag_update()
```

Comment:
This is the double call to flag_update() I referred to. It's already being called indirectly via `own.flag_update()`